src/Nox.Types.EntityFramework/Abstractions/NoxDatabaseConfigurator.cs

```diff
@@ -56,37 +56,71 @@ public abstract class NoxDatabaseConfigurator : INoxDatabaseConfigurator
-                    !relationshipToCreate.Relationship.IsManyRelationshipOnOtherSide())
+                // One to ?? (// Many to Many are setup by EF)
+                if (relationshipToCreate.Relationship.ShouldGenerateForeignOnThisSide() && relationshipToCreate.Relationship.WithSingleEntity())
                 {
-                    builder
-                        .HasOne(relationshipToCreate.Relationship.Entity)
-                        .WithOne(entity.Name)
-                        .HasForeignKey(entity.Name, $"{relationshipToCreate.Relationship.Entity}Id");
-
-                    // Setup one to one foreign key
+                    //One to One
+                    if (!relationshipToCreate.Relationship.IsManyRelationshipOnOtherSide())
+                    {
+//#if DEBUG2
+                        Console.WriteLine($"***Relationship oneToOne {entity.Name} ," +
+                            $"Name {relationshipToCreate.Relationship.Name} " +
+                            $"HasOne {relationshipToCreate.Relationship.Entity} " +
+                            $"WithOne {entity.Name}" +
+                            $"ForeignKey {relationshipToCreate.Relationship.Entity}Id " +
+                            $"");
+//#endif
+                        builder
+                            .HasOne(relationshipToCreate.Relationship.Entity)
+                            .WithOne(entity.Name)
+                            .HasForeignKey(entity.Name, $"{relationshipToCreate.Relationship.Entity}Id");
+                    }
+                    else //One to Many
+                    {
+//#if DEBUG
+                        Console.WriteLine($"***Relationship oneToMany {entity.Name}," +
+                           $"Name {relationshipToCreate.Relationship.Name} " +
+                           $"HasOne {$"{codeGeneratorState.DomainNameSpace}.{relationshipToCreate.Relationship.Entity}"} , {relationshipToCreate.Relationship.Entity} " +
+                           $"WithMany {entity.PluralName} " +
+                           $"ForeignKey {relationshipToCreate.Relationship.Entity}Id " +
+                           $"");
+//#endif
+
+                        builder
+                            .HasOne($"{codeGeneratorState.DomainNameSpace}.{relationshipToCreate.Relationship.Entity}", relationshipToCreate.Relationship.Entity)
+                            .WithMany(entity.PluralName)
+                            .HasForeignKey($"{relationshipToCreate.Relationship.Entity}Id");
+                    }
+                    // Setup foreign key property
                     ConfigureRelationForeignKeyProperty(codeGeneratorState, builder, entity, relationshipToCreate);
-                }
+                }               
 
                 if (!relationshipToCreate.Relationship.ShouldUseRelationshipNameAsNavigation())
                 {
+                    Console.WriteLine($"***Ignoring Navigation {relationshipToCreate.Relationship.Name}");
```

Comment:
same, do we want to have this?

Reply:
yes for now to help us troubleshoot anything, will remove before the release